src/websocket/stream.py

```diff
@@ -121,86 +132,104 @@ async def echo(self, websocket: WebSocket) -> None:
             self.close_stream = True
         finally:
             await websocket.close()
+            for task in self.transcription_tasks:
+                task.cancel()
```

Comment:
```suggestion
for task in self.transcription_tasks:
    task.cancel()

done, pending = await asyncio.wait(self.transcription_tasks, return_when=asyncio.ALL_COMPLETED)
self.transcription_tasks.difference_update(done)
```

**Why This Fix?** (This is including above fix in ensure_future)
add_done_callback(lambda t: self.transcription_tasks.discard(t)) ensures the task is removed only after it completes.
Using await asyncio.wait(self.transcription_tasks, return_when=asyncio.ALL_COMPLETED) ensures that all tasks finish before removing them.

Reply:
Good idea, however not relevant here. This just blocks the runner and the tasks would be terminated at some point anyway. I am not sure how waiting for the termination would benefit us